index.d.ts

```diff
@@ -0,0 +1,32 @@
+export interface IOptions {
+	/**
+	 * Directory to start from.
+	 *
+	 * @default process.cwd()
+	 */
+	cwd: string;
+	/**
+	 * [Normalize](https://github.com/npm/normalize-package-data#what-normalization-currently-entails) the package data.
+	 *
+	 * @default true
+	 */
+	normalize?: boolean;
+}
+
+export interface IPackage {
+	[key: string]: any;
+}
+
+/**
+ * Returns a `Promise` for the parsed JSON.
+ */
+declare function readPkg(options?: IOptions): Promise<IPackage>;
+
+declare namespace readPkg {
```

Comment:
In TS 3.1, we can now declare props on functions, so I don't think we need a namespace for this, see: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-1.html

Reply:
I'm trying to get this style of definition to work, but I'm not sure how to translate their example (which is for a regular .ts file) to work with a definition file with an ambient context, where you can't make assignments.

Reply:
Hmm, I think you're right. Actually, we can just export the `sync` method as a named export.

Reply:
Are you talking about like so?

```typescript
export default function (options?: Options): Promise<PackageMetadata>;
export function sync (options?: Options): PackageMetadata;
```

I believe that would require a change to the JS if that's the plan.

Reply:
Yup, exactly like that.